File: Examples/test-suite/csharp/director_binary_string_rev_len_runme.cs
using System;
using director_binary_string_rev_lenNamespace;

public class director_binary_string_rev_len_runme
{
    static void Main()
    {
      Caller caller = new Caller(new DirectorReverseLengthStringCallback());
      int sum = caller.call();

      if (sum != 9*2*8)
        throw new Exception("Unexpected sum: " + sum);
    }
}

public class DirectorReverseLengthStringCallback : Callback {
    public DirectorReverseLengthStringCallback() : base() {}

    public override int run(string dataBufferAA)
    {
      int ret = 0;
      if (dataBufferAA != null) {
        for (int i = 0; i < dataBufferAA.Length; i++)
          ret += (int)dataBufferAA[i] * 2;
      }
      return ret;
    }
}

File: Examples/test-suite/php/director_binary_string_rev_len_runme.php
<?php

require "tests.php";

// No new functions
check::functions(array());
check::classes(array('director_binary_string_rev_len','Callback','Caller'));
// No new vars
check::globals(array());

class DirectorBinaryStringCallback extends Callback {
  function run(string $dataBufferAA)
  {
    $ret = 0;
    if ($dataBufferAA != null) {
      for ($i = 0; $i < strlen($dataBufferAA); $i++)
        $ret = $ret + ord($dataBufferAA[$i]) * 2;
    }
    return $ret;
  }
}

$callback = new DirectorBinaryStringCallback();
$caller = new Caller($callback);
$sum = $caller->call();

check::equal($sum, 9*2*8, "Unexpected sum: $sum");
